pennylane/interfaces/batch/__init__.py

```diff
@@ -66,8 +66,7 @@ def set_shots(device, shots):
             device.shots = shots
         yield
     finally:
-        if device.shots != original_shots:
-            device.shots = original_shots

```

Comment:
@glassnotes this is the bug affecting the notebook - `!=` isn't safe enough, better to simply always revert the shots.